tests/ui/small_underscore_scope.rs

```diff
@@ -0,0 +1,27 @@
+// run-rustfix
+
+#![warn(clippy::small_underscore_scope)]
+
+struct NewType(u8);
+
+#[allow(dead_code)]
+struct Foo {
+    x: u8,
+    y: u8,
+}
+
+fn main() {
+    let n = NewType(24);
+    let NewType(_) = n;
+    let NewType(..) = n;
+
+    let t = (4, 5);
+    let (_, _) = t;
+    let (..) = t;
+
+    let t2 = (6, (7, 8));
+    let (_x, (_, _)) = t2;
+
+    let f = Foo { x: 3, y: 7 };
+    let Foo { .. } = f;
```

Comment:
Could you add a test for `println!("foo");`? It currently triggers on macro calls like [here](https://travis-ci.com/rust-lang/rust-clippy/jobs/194022153#L4206-L4210).

You'll probably need to add a `in_external_macro` check and return early in your lint code.